crates/api/src/transport.rs

```diff
@@ -169,6 +177,72 @@ impl TxImpHnd {
             Ok(())
         })
     }
+
+    /// Check whether a message should be rejected and the connection closed
+    /// due to all agents of the given peer being blocked and the message not
+    /// being one of the message types that are allowed anyway.
+    pub fn reject_message_if_all_agents_blocked(
+        &self,
+        peer: &Url,
+        space_id: &Option<Bytes>,
+        module_id: &Option<String>,
+        wire_type: &K2WireType,
+    ) -> K2Result<()> {
+        // We accept the following messages also for peers at whose url all
+        // agents are blocked:
+        //
+        // - Preflight: Such that we can discover any new agent infos available
+        //   at that peer URL (agent infos are sent via preflight messages and
+        //   a blocked agent cannot gossip, so we couldn't otherwise reliably
+        //    learn about new agents at a peer URL).
+        // - Unspecified: We allow unspecified messages in order to ensure that
+        //   we're not constraining us with regards to updates in the
+        //   networking protocol. And since we ignore unspecified messages
+        //   anyway, we wouldn't gain much from blocking.
+        // - Disconnect: If we receive a Disconnect message, we disconnect
+        //   anyway and a disconnect message also wouldn't include a space id
+        //   for which we could check for blocked agents.
+        if matches!(
+            wire_type,
+            K2WireType::Preflight
+                | K2WireType::Unspecified
+                | K2WireType::Disconnect
+        ) {
+            return Ok(());
+        }
```

Comment:
I think it would be simpler and easier to understand without this `matches!` and have `reject_message_if_all_agents_blocked()` get called within the `K2WireType::Notify` and `K2WireType::Module` branches in the higher function.